src/app/record/object-record/object-record.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { JsonPatch } from '../../shared/interfaces';
+
+@Component({
+  selector: 'object-record',
+  templateUrl: './object-record.component.html',
+  styleUrls: ['./object-record.component.scss']
+})
+export class ObjectRecordComponent implements OnInit {
+  @Input() record: object;
+  @Input() diff?: JsonPatch[];
+  @Input() depth: number;
+  @Input() expanded: boolean;
+
+  keys: Array<string> = [];
+  expandObject = false;
+  diffType: string;
+  private isArray = false;
+  isExpandedMap = new Map<string,boolean>();
+  diffKeys = new Map<string,boolean>();
+
+  ngOnInit() {
+    this.keys = Object.keys(this.record);
+    if (this.diff) {
+      this.diff.forEach(item => {
+        if (item.path.length > this.depth) {
+          this.diffKeys.set(item.path[this.depth],true);
+          this.expand(item.path[this.depth]);
+        } else if (item.path.length <= this.depth) {
+          this.expandObject = true;
+        }
+        if (item.path.length === (this.depth + 1) && item.op === 'add') {
+          this.record[item.path[this.depth]] = item.value;
+        }
+      }
+      )
+      if (this.expandObject) {
+        this.keys.forEach(item =>
+          {
+            this.diffKeys.set(item,true);
+        });//expand the added object
+        this.diffKeys.forEach((value, key) => { this.expand(key) });
+      }
+      this.diffType = this.diff[0].op;
+    }
+  }
+
+  trackByFunction(index: number, item: object): object {
+    return item;
+  }
+
+  getCorrespondingDiff(key: any) {
```

Comment:
duplicated this, so somehow should be shared with other class. Maybe you could create a base class to share this kind of things or have a service.